test/test_datapipe.py

```diff
@@ -569,6 +570,14 @@ def _return_self(x):
         # __len__ Test: returns the number of batches
         self.assertEqual(3, len(batch_dp))
 
+    def test_flatmap_datapipe(self):
```

Comment:
Could you please add a more comprehensive tests?
We at least need to add validation on `reset` using `reset_after_n_next_calls` shown from the other test cases, and `__len__` test.

Reply:
Yes, I'll add some tests momentarily.